packages/loopring_v3/contracts/core/iface/ExchangeData.sol

```diff
@@ -158,7 +158,7 @@ library ExchangeData
         uint[24]                 balanceMerkleProof;
     }
 
-    struct FlashMint
+    struct LoanDeposit
```

Comment:
Any specific reason to call this a LoanDeposit instead of a FlashMint? With FlashMint/FlashLoan being the standard name for this. And if you would explain what a LoanDeposit is (which probably nobody would call it when talking about it) you would simply say it's an L2 flash mint/loan. So seems easier and less confusing to just call it that directly without creating new terms?

I also like FlashMint because it contains the 2 most important characteristics:
- Flash: only usable within the same transaction
- Mint: the tokens aren't loaned from somewhere, they are really just created (which is why I chose FlashMint and not FlashLoan).

LoanDeposit does not make that clear, and the fact that internally a deposit is used is just an implementation detail.

Reply:
I thought you will ask this, sorry that I didn't explain it before.

Based on my understanding, the minted pool token will live more than one Ethreum transaction across multiple L2 blocks. It will really exist on L1 and L2, especially for the two-phase converters. Therefore, flash is misleading (at least it did mislead me). 

My understanding the following may be called a flash mint:

1. mint a token on L1
1. deposit it to L2
1. use it somehow on L2 but get it back to the same L2 account in the same Ethereum transaction
1. withdraw it to L1
1. burn it all on L1

All these actions need to be in one Ethreum block, so the new token can be known as "flash minted". Whereas our current design is more like:
**[loan ->deposit -> mint -> pay-loan]**

`amountFlashMinted` is past tense, but the tokens are yet to be minted. And I really think `amountFlashMinted` represents a loan. 

The repayFlashMint/repayDepositLoan also doesn't require the token need to be a newly minted token, it can be LRC or Ether, or any existing token. So *Mint* is misleading and *Flash* seems not accurate.

Let me know what you think.

Reply:
Oh okay I see what you're trying to say. But let's make sure we keep the two kinds of minting separate because like you said, they don't have anything to do with each other because flash mints work for all tokens.

a) The flash mint on L2 (exchange feature, nothing to do with converters)
b) The pool token minting in converters (external vault feature, nothing to do with the exchange)

> the minted pool token will live more than one Ethereum transaction across multiple L2 blocks. It will really exist on L1 and L2, especially for the two-phase converters. Therefore, flash is misleading (at least it did mislead me).

- The minted tokens for a) (the exchange level) only live for a single Ethereum tx. The flash minting feature only works on this level.
- The minted tokens for b) (external vault level) can exist for as long as needed. These don't directly have anything to do with the flash minting going on on the exchange.

Both should not be mixed together, they are completely separate mechanisms.

>My understanding the following may be called a flash mint:
>    1. mint a token on L1
>    2. deposit it to L2
>    3. use it somehow on L2 but get it back to the same L2 account in the same Ethereum transaction
>    4. withdraw it to L1
>    5. burn it all on L1

The only difference for our implementation is 5), because instead of burning the minted tokens, we actually want to keep them around and so we transfer them in to make them "real". So in that respect we have created some kind of hybrid between a flash mint and a flash loan. The tokens are actually minted like in a flash mint (they may not exist yet), but it's repaid like a flash loan.

> All these actions need to be in one Ethereum block, so the new token can be known as "flash minted". 

This is true in our implementation as well as the feature only impacts a).

> amountFlashMinted is past tense, but the tokens are yet to be minted.

That's not true. `amountFlashMinted` is only from the exchange's point of view, which is a) where the flash minting is happening. The minting has happened on the exchange and `amountFlashMinted` needs to be paid back in the same Ethereum tx. The exchange does not care where the tokens come from (minted in some converter or not).

Now the confusing thing is that for the converter there is also another kind of minting happening which is b). So indeed for b) the minting still needs to happen, but that has nothing to do with a). But it is a bit strange to understand because of how we use the converter: in a) we mint tokens where the total supply on the converter's side is 0!

> The repayFlashMint/repayDepositLoan also doesn't require the token need to be a newly minted token, it can be LRC or Ether, or any existing token. So Mint is misleading and Flash seems not accurate.

But they are minted because they don't need to exist yet (all deposits are actually mints from L2s point of view but that's beside the point). The converter is actually a good example why it's a mint and not a loan because the total supply in a converter is still 0 when we mint the tokens on the exchange! The tokens cannot be loaned from anywhere because they don't exist yet. We say some amount of tokens will exist so we have them immediately and will then prove it later by transferring them in.

So I would say:
- Flash is still very much what you would expect because it solely describes the mechanism on the exchange level 
- Mint makes it clear that they are minted initially, though you correctly point out that the repayment mechanism works like a flash loan.

Reply:
I see, all you said above make sense, but it does take some explanation which is not ideal for Bridge developers (the two different 'mint' is confusing). If you don't like the term "loanDeposit", how about we do not use 'mint', or 'loan' for a). We can change the names as follows:

- flashMint => flashDeposit
- repayFlashMint => repayFlashDeposit
- getDepositLoanAmount => getFlashDepositAmount
- amountFlashMinted => flashDepositAmounts

The names are cooler than loan***.

Reply:
I'm fine with flashDeposit! (this was actually my original name, so we're back to the start haha :).

> but it does take some explanation which is not ideal for Bridge developers (the two different 'mint' is confusing). 

Normally developers building on the Bridge/BaseConverter will not know about any of the minting going in in the exchange or even in the converter, that's all abstracted away.